e2e-tests/src/main.rs

```diff
@@ -92,26 +92,26 @@ pub struct RootArgs {
     #[arg(env = "E2E_OUTPUT_FORMAT", hide_env_values = true)]
     pub output_format: OutputFormat,
 
-    /// Scenarios to run
-    #[arg(long, value_name = "SCENARIO", value_enum, num_args = 1.., value_delimiter = ',')]
-    pub scenarios: Vec<Scenario>,
+    /// Protocols to run
+    #[arg(long, value_name = "PROTOCOL", value_enum, num_args = 1.., value_delimiter = ',')]
+    pub protocols: Vec<Protocol>,
 }
 
 #[derive(Debug, Clone, clap::ValueEnum, Copy)]
-pub enum Scenario {
+pub enum Protocol {
     Ethereum,
     Near,
     Stellar,
     Icp,
 }
 
-impl std::fmt::Display for Scenario {
+impl std::fmt::Display for Protocol {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         match self {
-            Scenario::Ethereum => write!(f, "ethereum"),
-            Scenario::Near => write!(f, "near"),
-            Scenario::Stellar => write!(f, "stellar"),
-            Scenario::Icp => write!(f, "icp"),
+            Protocol::Ethereum => write!(f, "ethereum"),
+            Protocol::Near => write!(f, "near"),
+            Protocol::Stellar => write!(f, "stellar"),
+            Protocol::Icp => write!(f, "icp"),
         }
     }
 }
```

Comment:
first, introduce a `Protocol::as_str` method which you can then call from here (if it's absolutely necessary, I'm not convinced it is)

Reply:
this is unresolved. drop `Protocol: fmt::Display` which returns a `String` for `Protocol::as_str()` which returns `&'static str`